orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/ServerGroupTagGenerator.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.servergroup;
+
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+
+import java.util.Map;
+
+public interface ServerGroupTagGenerator {
```

Comment:
Might suggest renaming this to `ServerGroupEntityTagGenerator` to be very explicit that this is for entity tags and not native asg tags.

Otherwise reasonable to me ... "some day" we might make this a little more generic with an `EntityTagGenerator` that could support generating tags for things other than server groups.